Gems/ROS2/Code/Source/RobotImporter/Utils/SourceAssetsStorage.cpp

```diff
@@ -353,11 +353,35 @@ namespace ROS2::Utils
                 if (outcomeCopyTmp)
                 {
                     // create asset info at destination location using the temporary mesh file
-                    const bool assetInfoOk =
-                        CreateSceneManifest(targetPathAssetTmp.String(), targetPathAssetInfo.String(), needsCollider, needsVisual);
+                    const bool assetInfoOk = CreateSceneManifest(targetPathAssetTmp, targetPathAssetInfo, needsCollider, needsVisual);
 
                     if (assetInfoOk)
                     {
+                        // copy additional assets such as textures directly to destination location
+                        const auto& assetPaths = Utils::GetMeshAssets(targetPathAssetTmp);
+                        for (const auto& unresolvedAssetFilepath : assetPaths)
+                        {
+                            // Manifest returns local path in Project's directory temp folder
+                            const AZ::IO::Path assetLocalPath(
+                                AZ::IO::Path(AZ::IO::Path(AZ::Utils::GetProjectPath()) / unresolvedAssetFilepath)
+                                    .LexicallyRelative(importDirectoryTmp));
+
+                            const AZ::IO::Path assetFullPathSrc(AZ::IO::Path(resolvedPath.ParentPath()) / assetLocalPath);
+                            const AZ::IO::Path assetFullPathDst(importDirectoryDst / assetLocalPath);
+
+                            const auto outcomeMkdir = fileIO->CreatePath(AZ::IO::Path(assetFullPathDst.ParentPath()).c_str());
+                            if (!outcomeMkdir)
+                            {
+                                break;
+                            }
+
+                            const auto outcomeCopy = fileIO->Copy(assetFullPathSrc.c_str(), assetFullPathDst.c_str());
+                            if (outcomeCopy)
+                            {
+                                copiedFiles[assetFullPathSrc.String()] = assetFullPathDst.String();
+                            }
+                        }
```

Comment:
I have a feeling this logic is going to need some iteration over time, but I'm OK with it going in and improving it as we hit issues. Some of the things I'm thinking about:
* If the mesh refers to textures that can't be found, this doesn't look like it's doing much in the way of error-detection and reporting yet.
* @lumberyard-employee-dm We might hit an odd problem with some upcoming work, where we add URI prefix resolution for material textures to the asset import. For the URI prefix resolution to work, it needs to be able to find the texture. If we try to do the resolution before copying the textures, the textures won't exist anywhere that we can find them yet. If we try to copy the textures first, we won't have done the prefix resolution yet to know how to copy them. It's possible we'll need to do it in both places... :(